File: java to c++/Responses/TFTPError.cpp
#include "TFTPError.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace Commands
					{
						namespace Responses
						{

							std::wstring TFTPError::getErrorMsg()
							{
								return errorMsg;
							}

							short TFTPError::getErrorCode()
							{
								return errorCode;
							}

							TFTPError::TFTPError(short errorCode, const std::wstring &errorMsg)
							{
								this->errorMsg = errorMsg;
								this->errorCode = errorCode;
							}

							TFTPError::TFTPError(short errorCode)
							{
								this->errorMsg = errorMessages[errorCode];
								this->errorCode = errorCode;
							}

							TFTPError::TFTPError(int errorCode) : TFTPError((short)errorCode)
							{
							}

							short TFTPError::getOpCode()
							{
								return static_cast<short>(5);
							}

							std::wstring TFTPError::errorMessage()
							{
								return errorMsg;
							}

std::vector<std::wstring> TFTPError::errorMessages = {L"Not defined, see error message (if any).", L"File not found", L"Access violation", L"Disk full or allocation exceeded", L"Illegal TFTP operation", L"File already exists", L"User not logged in", L"User already logged in"};
						}
					}
				}
			}
		}
	}
}

File: Server/src/main/java/bgu/spl171/net/impl/TFTP/Commands/Responses/TFTPError.java
package bgu.spl171.net.impl.TFTP.Commands.Responses;

public class TFTPError implements Response {
    String errorMsg;
    short errorCode;

    public String getErrorMsg() {
        return errorMsg;
    }

    public short getErrorCode() {
        return errorCode;
    }

    public TFTPError(short errorCode, String errorMsg) {
        this.errorMsg = errorMsg;
        this.errorCode = errorCode;
    }

    public TFTPError(short errorCode) {
        this.errorMsg = errorMessages[errorCode];
        this.errorCode = errorCode;
    }
    public TFTPError(int errorCode){
        this((short)errorCode);
    }

    @Override
    public short getOpCode() {
        return (short) 5;
    }

    public String errorMessage() {
        return errorMsg;
    }

    private static String[] errorMessages = {
            "Not defined, see error message (if any).",
            "File not found",
            "Access violation",
            "Disk full or allocation exceeded",
            "Illegal TFTP operation",
            "File already exists",
            "User not logged in",
            "User already logged in"
    };
}
